portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -116,19 +123,34 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     Date datetime = new Date();
 
     // Check for validity.
-    if (!body.isEmpty()) {
-      // Buld the new comment.
-      Entity commentEntity = new Entity("Comment");
-      commentEntity.setProperty("author", author);
-      commentEntity.setProperty("body", body);
-      commentEntity.setProperty("datetime", datetime);
-
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-      datastore.put(commentEntity);
+    if (body.isEmpty()) {
+      response.setStatus(400);
+      return;
     }
 
-    // Redirect back to the Contact page.
-    response.sendRedirect("/#/contact");
+    // Check the sentiment of the comment entered
+    Document doc = Document.newBuilder().setContent(body).setType(Document.Type.PLAIN_TEXT).build();
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+    float score = sentiment.getScore();
+    languageService.close();
+
+    response.setStatus(200);
+    response.getWriter().println(score);
+
+    if (score < -0.7) {
```

Comment:
You can make this a constant for better readability eg. `SENTIMENT_THRESHOLD`

Reply:
Okay, sounds good!